### PR TITLE
add spaces between groups of digits

### PR DESCRIPTION
**Pull request by mikehaverstock**
### Testing:
`make server`
Then open http://localhost:9292/test/digit-grouping.html

### TODO:
- [x] put grouping behind a flag
- [x] find all edge cases where we don't run the grouping algorithm (I think we need to listen for DOT and SPACE nodes being inserted next to NON digits.
- [x] call the grouping algorithm from the optimization for simple number mutations
- [x] fix grouping algorithm when running in renderLatexMathEfficiently() - the demo page points out a problem
- [x] optimize removeClass() and addClass() calls
- [x] decide if we want to style the manual spaces as thin spaces. I'm hoping the answer is no, but could be persuaded it's a good idea.
- [x] add tests

### CODE REVIEW

**Comment by mikehaverstock** [comment]
@ehberger - I need to add some tests, but would love some feedback on this before writing a bunch of tests. Does this feel like it implements the spec correctly?

**Comment by mikehaverstock** [comment]
@eluberoff or @jwmerrill - this is finished. Not sure who is best to review it

File: src/commands/math/basicSymbols.js [hunk 1]
```diff
@@ -1,8 +1,160 @@
...
+  _.fixDigitGrouping = function (opts) {
+    if (!opts.enableDigitGrouping) return;
+
+    var left = this;
+    var right = this;
+
+    var spacesFound = 0;
+    var dots = [];
+
+    var SPACE = '\\ ';
```
Comment (jwmerrill) [inline src/commands/math/basicSymbols.js]: There are a lot of different kinds of spacing you can use in LaTeX:

https://tex.stackexchange.com/questions/74353/what-commands-are-there-for-horizontal-spacing/74354#74354

I don't know which of these mathquill supports, and I don't think we give you any ways to enter any of them other than `\ ` in the calculator, but just wanted to call this out as a case we might be missing.

File: src/commands/math/basicSymbols.js [hunk 2]
```diff
@@ -1,8 +1,160 @@
...
+      } else if (node.ctrlSeq === SPACE) {
+        right = node
+        spacesFound += 1;
+      } else if (node.ctrlSeq === DOT) {
+        right = node
+        dots.push(node);
+      } else {
+        break;
+      }
+    }
```
Comment (jwmerrill) [inline src/commands/math/basicSymbols.js]: Why is left a `do...while` loop, and right is only a `while` loop?

File: test/demo.html [hunk 3]
```diff
@@ -93,6 +93,7 @@ <h1><a href="http://mathquill.github.com">MathQuill</a> Demo <small>local test p
   $(latexMath.el()).bind('keydown keypress', function() {
     setTimeout(function() {
       var latex = latexMath.latex();
+
```
Comment (jwmerrill) [inline test/demo.html]: Unnecessary edit

File: test/unit/digit-grouping.test.js [hunk 4]
```diff
@@ -0,0 +1,1172 @@
...
+                    content: '5'
+                  },
+                ]
+            }
+          })
+          done();
+        }, 1);
+      }, 1100); // should stop suppressing grouping after 1000ms
+    });
+});
```
Comment (jwmerrill) [inline test/unit/digit-grouping.test.js]: Really nice set of tests

File: src/commands/math/basicSymbols.js (hunk 2 above)
Comment (mikehaverstock) [reply src/commands/math/basicSymbols.js]: It's because the 'left' side starts at the original character and the 'right' side starts to the right of the original character. We want to always process the original character in the left loop. There may or may not be a character to the right. If this is confusing, I can rewrite it, but I don't think it'll make it any less confusing.
